sdk/identity/Azure.Identity/src/AzureCliCredentialClient.cs

```diff
@@ -0,0 +1,172 @@
+        }
+
+        public virtual async ValueTask<AccessToken> RequestCliAccessTokenAsync(string[] scopes, CancellationToken cancellationToken)
+        {
+            return await RequestCliAccessTokenAsync(true, scopes, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected virtual async ValueTask<AccessToken> RequestCliAccessTokenAsync(bool async, string[] scopes, CancellationToken cancellationToken)
+        {
+            string resource = ScopeUtilities.ScopesToResource(scopes);
+
+            ScopeUtilities.ValidateScope(resource);
+
+            (string output, int exitCode) = await GetAzureCliAccesToken(async, resource, cancellationToken).ConfigureAwait(false);
+
+            if (exitCode != 0)
+            {
+                bool isLoginError = output.StartsWith("Please run 'az login'", StringComparison.CurrentCultureIgnoreCase);
+                bool isWinError = output.StartsWith(WinAzureCLIError, StringComparison.CurrentCultureIgnoreCase);
+                string pattter = "az:(.*)not found";
+                bool isOtherOsError = Regex.IsMatch(output, pattter);
+
+                if (isWinError || isOtherOsError)
+                {
+                    throw new CredentialUnavailableException(AzureCLINotInstalled);
+                }
+                else if (isLoginError)
+                {
+                    throw new CredentialUnavailableException(AzNotLogIn);
+                }
+
+                throw new AuthenticationFailedException(output);
+            }
+
+            byte[] byteArrary = Encoding.ASCII.GetBytes(output);
+
+            using MemoryStream stream = new MemoryStream(byteArrary);
+
+            return await AccessTokenUtilities.DeserializeAsync(async, stream, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected virtual async ValueTask<(string, int)> GetAzureCliAccesToken(bool async, string resource, CancellationToken cancellationToken)
+        {
+            string command = $"az account get-access-token --output json --resource {resource}";
+
+            string fileName = string.Empty;
+            string argument = string.Empty;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                fileName = "cmd";
+                argument = $"/c \"{command}\"";
+            }
+            else
+            {
+                fileName = "/bin/sh";
+                argument = $"-c \"{command}\"";
+            }
+
+            ProcessStartInfo procStartInfo = new ProcessStartInfo()
+            {
+                FileName = fileName,
+                Arguments = argument,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+                WorkingDirectory = _workingDir,
+                Environment = { { "PATH", _path } }
```

Comment:
Does this mean that credential would only work with `az` installed into a default location?